src/adic_space.lean

```diff
@@ -8,11 +8,132 @@ import for_mathlib.topology
+    tendsto_infi' ((λ r, -r) '' V)
+      begin 
+        simp [nhd_zero_symmetric h],
+        unfold tendsto,
+        rw map_principal,
+        have H : prod.swap '' {p : R × R | ∃ (x : R), x ∈ V ∧ -x = p.snd + -p.fst} = {p : R × R | p.snd - p.fst ∈ V},
+        begin 
+          rw set.image_swap_eq_preimage_swap,
+          ext p,
+          cases p with r1 r2,
+          change (∃ x, x ∈ V ∧ -x = r1 - r2) ↔ (r2 - r1 ∈ V),
+          have : ∀ x, -x = r1 - r2 ↔ x = r2 - r1,
+            by intro x; rw [neg_eq_iff_neg_eq, eq_comm]; simp,
+          simp only [this], simp
+      end,
+      simp [H],                           
+    end,
+  comp        := le_infi $ assume V, le_infi $ assume H, 
+    begin
+      rcases half_nhd V H with ⟨U, ⟨h, h'⟩⟩,
+      have comprel : comp_rel {p : R × R | p.snd - p.fst ∈ U} {p : R × R | p.snd - p.fst ∈ U} ⊆ {p : R × R | p.snd - p.fst ∈ V},
+      begin
+        intros p p_comp_rel,
+        rcases p_comp_rel with ⟨z, ⟨Hz1, Hz2⟩⟩,
+        simpa using h' _ _ Hz1 Hz2,
+      end,
+      replace comprel : principal (comp_rel {p : R × R | p.snd - p.fst ∈ U} {p : R × R | p.snd - p.fst ∈ U}) ≤ principal {p : R × R | p.snd - p.fst ∈ V},
+        simpa using comprel,
+      apply lift'_le,
+      swap,
+      exact comprel,
+      apply mem_infi_sets U,
+      apply mem_infi_sets h,
+      simp
+    end }
+
+variable (R)
+/-- Wedhorn Definition 5.31 page 38 -/
+definition is_complete : Prop := is_hausdorff R ∧ ∀ {f:filter R}, cauchy f → ∃x, f ≤ nhds x
+
+variable {R}
+/-- Wedhorn Definition 5.27 page 36 -/
+definition is_bounded 
+  (B : set R) : Prop := ∀ U ∈ nhd_zero R, ∃ V ∈ nhd_zero R, ∀ v ∈ V, ∀ b ∈ B, v*b ∈ U
+
+def power_set (r : R) : set R := set.range (λ n : ℕ, r^n)
+
+definition is_power_bounded (r : R) : Prop := is_bounded (power_set r)
+
+variable (R)
+-- Somehow we need R° both as a subset of R and a subtype. 
+-- There is a coercion from the set to the subtype but relying naively on it seems to bring 
+-- type class resolution issues
+definition power_bounded_subring := {r : R // is_power_bounded r}
+definition power_bounded_subring_set := {r : R | is_power_bounded r}
+
+instance power_bounded_subring_to_ring : has_coe (power_bounded_subring R) R := ⟨subtype.val⟩ 
+instance power_bounded_subring_is_ring  : comm_ring (power_bounded_subring R) := sorry
+instance : topological_space (power_bounded_subring R) := sorry
+instance : topological_ring (power_bounded_subring R) := sorry
+
+definition is_uniform : Prop := is_bounded (power_bounded_subring_set R)
+
+theorem p_is_power_bounded [p : Prime] : is_power_bounded (p : power_bounded_subring R) := sorry
+
+variable {R}
+definition is_pseudo_uniformizer : R → Prop := sorry
+end topological_ring
 
 -- Schol= : "Recall that a topological ring R is Tate if it contains an
 -- open and bounded subring R0 ⊂ R and a topologically nilpotent unit pi ∈ R; such elements are
--- called pseudo-uniformizers.""
+-- called pseudo-uni7formizers.""
```

Comment:
Something went wrong here?

Reply:
Schol= -> Scholze as well (this was my fault)